JIT build failure with clang 10

This prevents godbolt from building nightly dotnet images.
The build has failed since June 16.

See https://github.com/compiler-explorer/compiler-workflows/actions/workflows/build-daily-dotnet.yml

```
    FAILED: jit/static/CMakeFiles/clrjit_static.dir/cmake_pch.hxx.pch 
    /usr/bin/clang++-10 -DDEBUGGING_SUPPORTED -DDISABLE_CONTRACTS -DFEATURE_BASICFREEZE -DFEATURE_CODE_VERSIONING -DFEATURE_COLLECTIBLE_TYPES -DFEATURE_COMWRAPPERS -DFEATURE_CORECLR -DFEATURE_DBGIPC_TRANSPORT_DI -DFEATURE_DBGIPC_TRANSPORT_VM -DFEATURE_DEFAULT_INTERFACES -DFEATURE_EH_FUNCLETS -DFEATURE_EVENTSOURCE_XPLAT -DFEATURE_EVENT_TRACE -DFEATURE_HIJACK -DFEATURE_HW_INTRINSICS -DFEATURE_ICASTABLE -DFEATURE_INSTANTIATINGSTUB_AS_IL -DFEATURE_MANUALLY_MANAGED_CARD_BUNDLES -DFEATURE_MASKED_HW_INTRINSICS -DFEATURE_METADATA_UPDATER -DFEATURE_MULTICOREJIT -DFEATURE_MULTIREG_RETURN -DFEATURE_ON_STACK_REPLACEMENT -DFEATURE_PAL_ANSI -DFEATURE_PERFMAP -DFEATURE_PERFTRACING -DFEATURE_PGO -DFEATURE_PORTABLE_SHUFFLE_THUNKS -DFEATURE_PROFAPI_ATTACH_DETACH -DFEATURE_READYTORUN -DFEATURE_REJIT -DFEATURE_REMAP_FUNCTION -DFEATURE_REMOTE_PROC_MEM -DFEATURE_SIMD -DFEATURE_STANDALONE_GC -DFEATURE_STUBS_AS_IL -DFEATURE_SVR_GC -DFEATURE_SYMDIFF -DFEATURE_TIERED_COMPILATION -DFEATURE_USE_ASM_GC_WRITE_BARRIERS -DFEATURE_USE_SOFTWARE_WRITE_WATCH_FOR_GC_HEAP -DHOST_64BIT -DHOST_AMD64 -DHOST_UNIX -DJIT_BUILD -DNDEBUG -DPROFILING_SUPPORTED -DTARGET_64BIT -DTARGET_AMD64 -DTARGET_LINUX -DTARGET_UNIX -DUNICODE -DUNIX_AMD64_ABI -DUNIX_AMD64_ABI_ITF -DURTBLDENV_FRIENDLY=Retail -D_FILE_OFFSET_BITS=64 -D_SECURE_SCL=0 -D_TIME_BITS=64 -D_UNICODE -I/root/dotnet/runtime/artifacts/obj/coreclr/linux.x64.Release/jit/static -I/root/dotnet/runtime/src/coreclr/jit/static -I/root/dotnet/runtime/src/native -I/root/dotnet/runtime/src/native/inc -I/root/dotnet/runtime/src/coreclr/pal/prebuilt/inc -I/root/dotnet/runtime/artifacts/obj -I/root/dotnet/runtime/src/coreclr/pal/inc -I/root/dotnet/runtime/src/coreclr/pal/inc/rt -I/root/dotnet/runtime/src/coreclr/pal/src/safecrt -I/root/dotnet/runtime/src/coreclr/inc -I/root/dotnet/runtime/src/coreclr/debug/inc -I/root/dotnet/runtime/src/coreclr/debug/inc/amd64 -I/root/dotnet/runtime/src/coreclr/debug/inc/dump -I/root/dotnet/runtime/src/coreclr/md/inc -I/root/dotnet/runtime/src/coreclr/classlibnative/bcltype -I/root/dotnet/runtime/src/coreclr/classlibnative/inc -I/root/dotnet/runtime/artifacts/obj/coreclr/linux.x64.Release/inc -I/root/dotnet/runtime/src/coreclr/hosts/inc -I/root/dotnet/runtime/src/coreclr/minipal -I/root/dotnet/runtime/src/coreclr/nativeresources -I/root/dotnet/runtime/src/coreclr/jit/./jitstd -I/root/dotnet/runtime/src/coreclr/jit/../inc -I/root/dotnet/runtime/src/coreclr/jit -O3 -DNDEBUG -std=gnu++11 -fPIC -O3 -Wall -Wno-null-conversion -glldb -fno-omit-frame-pointer -fwrapv -fstack-protector-strong -Werror -Wno-unused-variable -Wno-unused-value -Wno-unused-function -Wno-tautological-compare -Wno-unknown-pragmas -Wimplicit-fallthrough -Wvla -Wno-invalid-offsetof -Wno-unused-but-set-variable -ffp-contract=off -fno-rtti -Wno-unknown-warning-option -ferror-limit=4096 -Wno-unused-private-field -Wno-constant-logical-operand -Wno-pragma-pack -Wno-incompatible-ms-struct -Wno-reserved-identifier -Wno-unsafe-buffer-usage -Wno-single-bit-bitfield-constant-conversion -Wno-cast-function-type-strict -Wno-switch-default -fsigned-char -fvisibility=hidden -ffunction-sections -Wno-null-arithmetic -Wno-sync-alignment -Winvalid-pch -Xclang -emit-pch -Xclang -include -Xclang /root/dotnet/runtime/artifacts/obj/coreclr/linux.x64.Release/jit/static/CMakeFiles/clrjit_static.dir/cmake_pch.hxx -x c++-header -MD -MT jit/static/CMakeFiles/clrjit_static.dir/cmake_pch.hxx.pch -MF jit/static/CMakeFiles/clrjit_static.dir/cmake_pch.hxx.pch.d -o jit/static/CMakeFiles/clrjit_static.dir/cmake_pch.hxx.pch -c /root/dotnet/runtime/artifacts/obj/coreclr/linux.x64.Release/jit/static/CMakeFiles/clrjit_static.dir/cmake_pch.hxx.cxx
    In file included from <built-in>:1:
    In file included from /root/dotnet/runtime/artifacts/obj/coreclr/linux.x64.Release/jit/static/CMakeFiles/clrjit_static.dir/cmake_pch.hxx:5:
    In file included from /root/dotnet/runtime/src/coreclr/jit/jitpch.h:44:
    In file included from /root/dotnet/runtime/src/coreclr/jit/compiler.h:29:
    /root/dotnet/runtime/src/coreclr/jit/gentree.h:4660:10: error: reference to scoped enumeration must use 'enum' not 'enum class'
        enum class WellKnownArg WellKnownArg = ::WellKnownArg::None;
             ^
    1 error generated.
```

/cc: @EgorBo

--- a/src/coreclr/jit/gentree.h
+++ b/src/coreclr/jit/gentree.h
@@ -4658,7 +4658,7 @@ struct NewCallArg
     // The class handle if SignatureType == TYP_STRUCT.
     CORINFO_CLASS_HANDLE SignatureClsHnd = NO_CLASS_HANDLE;
     // The type of well known arg
-    enum class WellKnownArg WellKnownArg = ::WellKnownArg::None;
+    enum WellKnownArg WellKnownArg = ::WellKnownArg::None;
 
     NewCallArg WellKnown(::WellKnownArg type) const
     {

